File: articles/personality-work-and-leadership/fortran/work_leadership_summary.f90
program work_leadership_summary
  implicit none

  character(len=4096) :: line
  character(len=256) :: field
  integer :: unit, ios, n, pos, next_pos, col
  real :: performance, performance_sum

  performance_sum = 0.0
  n = 0

  open(newunit=unit, file="../data/synthetic_personality_work_leadership.csv", status="old", action="read", iostat=ios)
  if (ios /= 0) then
    print *, "Could not open data file"
    stop 1
  end if

  read(unit, '(A)', iostat=ios) line

  do
    read(unit, '(A)', iostat=ios) line
    if (ios /= 0) exit

    pos = 1
    col = 1
    performance = 0.0

    do
      next_pos = index(line(pos:), ",")
      if (next_pos == 0) then
        field = adjustl(line(pos:))
      else
        field = adjustl(line(pos:pos+next_pos-2))
      end if

      if (col == 13) read(field, *) performance

      if (next_pos == 0) exit
      pos = pos + next_pos
      col = col + 1
    end do

    performance_sum = performance_sum + performance
    n = n + 1
  end do

  close(unit)

  open(newunit=unit, file="../outputs/fortran_work_leadership_summary.txt", status="replace", action="write")
  write(unit, '(A,I0)') "n=", n
  write(unit, '(A,F8.4)') "job_performance_mean=", performance_sum / real(n)
  close(unit)

  print *, "Wrote Fortran output: ../outputs/fortran_work_leadership_summary.txt"
end program work_leadership_summary

File: articles/personality-work-and-leadership/c/work_leadership_summary.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LINE 4096

int main(void) {
    FILE *fp = fopen("../data/synthetic_personality_work_leadership.csv", "r");
    if (!fp) {
        perror("Could not open data file");
        return 1;
    }

    char line[MAX_LINE];
    if (!fgets(line, sizeof(line), fp)) {
        fprintf(stderr, "Missing header\n");
        fclose(fp);
        return 1;
    }

    int n = 0;
    double performance_sum = 0.0;

    while (fgets(line, sizeof(line), fp)) {
        char *token;
        int col = 0;
        token = strtok(line, ",");
        while (token != NULL) {
            if (col == 12) {
                performance_sum += atof(token);
            }
            token = strtok(NULL, ",");
            col++;
        }
        n++;
    }

    fclose(fp);

    if (n == 0) {
        fprintf(stderr, "No rows\n");
        return 1;
    }

    FILE *out = fopen("../outputs/c_work_leadership_summary.txt", "w");
    if (!out) {
        perror("Could not open output file");
        return 1;
    }

    fprintf(out, "n=%d\njob_performance_mean=%.4f\n", n, performance_sum / n);
    fclose(out);

    printf("Wrote C output: ../outputs/c_work_leadership_summary.txt\n");
    return 0;
}
